bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -134,15 +136,19 @@ void ReplicaImp::messageHandler(MessageBase *msg) {
       return;
     }
   }
-  if (validateMessage(trueTypeObj) && !isCollectingState())
+  if constexpr (std::is_same_v<T, PrePrepareMsg>) {
```

Comment:
You cannot handle any message while state transfer. Please add here this condition
```
if(!isCollectingState())
```

Reply:
Thanks @yontyon for your comment. Let me add this check.